lib/compress/zstd_fast.c

```diff
@@ -463,8 +505,11 @@ size_t ZSTD_compressBlock_fast_dictMatchState_generic(
     while (ip1 <= ilimit) {   /* repcode check at (ip0 + 1) is safe because ip0 < ip1 */
         size_t mLength;
         size_t hash0 = ZSTD_hashPtr(ip0, hlog, mls);
-        const size_t dictHash0 = ZSTD_hashPtr(ip0, dictHLog, mls);
-        U32 dictMatchIndex = dictHashTable[dictHash0];
+
+        size_t const dictHashAndTag0 = ZSTD_hashPtr(ip0, dictHLog + ZSTD_SHORT_CACHE_TAG_BITS, mls);
+        U32 dictMatchIndexAndTag = dictHashTable[dictHashAndTag0 >> ZSTD_SHORT_CACHE_TAG_BITS];
+        size_t dictTagsMatch = (dictMatchIndexAndTag & ZSTD_SHORT_CACHE_TAG_MASK) == (dictHashAndTag0 & ZSTD_SHORT_CACHE_TAG_MASK);
```

Comment:
nit: const. Also the 2nd line should be simplified. I'd recommend just adding a helper function to compare the hash/index tags.

Reply:
Which variable should be const? Agree on simplifying the dictTagsMatch logic.

Reply:
shouldn't both be? Or are they modified later?

Reply:
Yes, modified at the end of the loop:
```
            /* Prepare for next iteration */
            dictMatchIndexAndTag = dictHashTable[dictHashAndTag1 >> ZSTD_SHORT_CACHE_TAG_BITS];
            dictTagsMatch = (dictMatchIndexAndTag & ZSTD_SHORT_CACHE_TAG_MASK) == (dictHashAndTag1 & ZSTD_SHORT_CACHE_TAG_MASK);
```